sway/commands.c

```diff
@@ -169,6 +170,72 @@ static bool cmd_exit(struct sway_config *config, int argc, char **argv) {
 	return true;
 }
 
+static bool cmd_floating(struct sway_config *config, int argc, char **argv) {
+	if (strcasecmp(argv[0], "toggle") == 0) {
+		swayc_t *view = get_focused_container(&root_container);
+		// Prevent running floating commands on things like workspaces
+		if (view->type != C_VIEW) {
+			return true;
+		}
+		int i;
+		// Change from nonfloating to floating
+		if (!view->is_floating) {
+			view->is_floating = true;
+			for (i = 0; i < view->parent->children->length; i++) {
+				if (view->parent->children->items[i] == view) {
+					// Cut down on width/height so it's obvious that you've gone floating
+					// if this is the only view
+					view->width = view->width - 30;
+					view->height = view->height - 30;
+
+					// Swap from the list of whatever container the view was in
+					// to the workspace->floating list
+					// TODO: Destroy any remaining empty containers
+					list_del(view->parent->children, i);
+					list_add(active_workspace->floating, view);
```

Comment:
We should probably do this before we merge this PR. Empty containers being left around causes problems.